hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/upgrade/UpgradeDowngradeUtils.java

```diff
@@ -107,4 +132,95 @@ static void updateMetadataTableVersion(HoodieEngineContext context, HoodieTableV
       throw new HoodieIOException("Error while updating metadata table version", e);
     }
   }
+
+  static void upgradeToLSMTimeline(HoodieTable table, HoodieEngineContext engineContext, HoodieWriteConfig config) {
+    // migrate archived timeline
+    LSMTimelineWriter lsmTimelineWriter = LSMTimelineWriter.getInstance(config, table);
+    HoodieArchivedTimeline archivedTimeline = table.getMetaClient().getArchivedTimeline();
+    Consumer<Exception> exceptionHandler = e -> {
+      if (config.isFailOnTimelineArchivingEnabled()) {
+        throw new HoodieException(e);
+      }
+    };
+    lsmTimelineWriter.write(Collections.singletonList(ActiveAction.fromInstants(archivedTimeline.getInstants())),
```

Comment:
The list for archived timeline is huge, we need to split the list into small batchs and write into the LSM timeline per-batch, by default, just use 10 instants as a batch which is in line with the current behavior.

Reply:
Another concern is that the legacy archived timeline may contain enormous instants there (like several GBs of avro logs), it would be very time-consuming to load the whole legacy archived timeline, maybe we just load the latest avro log(which should be enough for file slicing).